server/src/main/java/org/elasticsearch/common/geo/SimpleVectorTileFormatter.java

```diff
@@ -36,28 +45,37 @@ public String getName() {
     }
 
     /**
-     * Parses string in the format we expect either z/x/y or z/x/y@extent to an array of integer parameters
+     * Parses string in the format we expect, primarily z/x/y, to an array of integer parameters.
+     * There are also two optional additions to the format:
+     * <ul>
+     *     <li><code>@extent</code> - number of pixels across</li>
+     *     <li><code>:buffer</code> - number of pixels by which to widen each tile</li>
+     * </ul>
+     * The resulting array will always have five elements, with the extent and buffer set to the default
+     * values if not provided.
+     * Examples:
+     * <ul>
+     *     <li><code>3/2/2</code> produces <code>[3,2,2,4096,5]</code></li>
+     *     <li><code>2/1/1@5000</code> produces <code>[2,1,1,5000,5]</code></li>
+     *     <li><code>2/1/1@5000:10</code> produces <code>[2,1,1,5000,10]</code></li>
+     * </ul>
      */
     public static int[] parse(String param) {
-        // we expect either z/x/y or z/x/y@extent
-        final String[] parts = param.split("@", 3);
-        if (parts.length > 2) {
+        Matcher matcher = pattern.matcher(param);
+        if (matcher.matches() == false) {
             throw new IllegalArgumentException(
                 "Invalid mvt formatter parameter [" + param + "]. Must have the form \"zoom/x/y\" or \"zoom/x/y@extent\"."
             );
         }
-        final int extent = parts.length == 2 ? Integer.parseInt(parts[1]) : 4096;
-        final String[] tileBits = parts[0].split("/", 4);
-        if (tileBits.length != 3) {
-            throw new IllegalArgumentException("Invalid tile string [" + parts[0] + "]. Must be three integers in a form \"zoom/x/y\".");
-        }
-        final int z = GeoTileUtils.checkPrecisionRange(Integer.parseInt(tileBits[0]));
+        final int z = GeoTileUtils.checkPrecisionRange(Integer.parseInt(matcher.group(1)));
         final int tiles = 1 << z;
-        final int x = Integer.parseInt(tileBits[1]);
-        final int y = Integer.parseInt(tileBits[2]);
+        final int x = Integer.parseInt(matcher.group(2));
+        final int y = Integer.parseInt(matcher.group(3));
         if (x < 0 || y < 0 || x >= tiles || y >= tiles) {
             throw new IllegalArgumentException(String.format(Locale.ROOT, "Zoom/X/Y combination is not valid: %d/%d/%d", z, x, y));
         }
-        return new int[] { z, x, y, extent };
+        final int extent = matcher.group(4) == null ? DEFAULT_EXTENT : Integer.parseInt(matcher.group(4).substring(1));
+        final int bufferPixels = matcher.group(5) == null ? DEFAULT_BUFFER_PIXELS : Integer.parseInt(matcher.group(5).substring(1));
+        return new int[] { z, x, y, extent, bufferPixels };
```

Comment:
Now that we are here, we can check that extent is greater than 0 and buffer pixels greater or equal to 0?

Reply:
Great idea. Since this method is doing validation anyway, it makes sense for it to do more validation.

Reply:
Since the regex already does not accept any negative numbers, the only missing check was on `extent==0`, so I'm adding that. One consequence of the regex is that most invalid content will result in the same error message, while only the `extent==0` is a more specific error message.